### PR TITLE
Elasticsearch: Remove support for versions after their end of the life (<7.10.0)

### LINKED ISSUE #48496: [9.0] Elastic - support only versions that are still in support with Elastic

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
In https://github.com/grafana/grafana/pull/48506 we have added deprecation notice and for v9 we are removing support for 
Elasticsearch data sources with versions < 7.10 as they are past EOL based on https://www.elastic.co/support/eol.

image

**Which issue(s) this PR fixes**:

Fixes: https://github.com/grafana/grafana/issues/48496

# Release notice breaking change
Removes support for for ElasticSearch versions after their end-of-life, currently versions < 7.10.0.  To continue to use ElasticSearch data source, upgrade ElasticSearch to version 7.10.0+.

### CODE REVIEW

File: pkg/tsdb/elasticsearch/elasticsearch.go [hunk 1]
```diff
@@ -36,15 +36,21 @@ func ProvideService(httpClientProvider httpclient.Provider) *Service {
...
-
 	dsInfo, err := s.getDSInfo(req.PluginContext)
 	if err != nil {
 		return &backend.QueryDataResponse{}, err
 	}
 
+	// Support for version after their end-of-life (currently <7.10.0) was removed
+	lastSupportedVersion, _ := semver.NewVersion("7.10.0")
+	if dsInfo.ESVersion.LessThan(lastSupportedVersion) {
+		return &backend.QueryDataResponse{}, fmt.Errorf("support for elasticsearch versions after their end-of-life (currently versions < 7.10) was removed, err=%v", err)
```
Comment (gabor) [inline pkg/tsdb/elasticsearch/elasticsearch.go:47]: i think there's a problem. the `err` variable you use here is coming from the `dsInfo, err := s.getDsInfo...` call above..  i think you can just have 
```suggestion
		return &backend.QueryDataResponse{}, fmt.Errorf("support for elasticsearch versions after their end-of-life (currently versions < 7.10) was removed")
```

**Review by meanmina — APPROVED** [review]
LGTM 👍

**Review by chri2547 — APPROVED** [review]

File: pkg/tsdb/elasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -36,15 +36,21 @@ func ProvideService(httpClientProvider httpclient.Provider) *Service {
...
 	}
 
+	// Support for version after their end-of-life (currently <7.10.0) was removed
+	lastSupportedVersion, _ := semver.NewVersion("7.10.0")
+	if dsInfo.ESVersion.LessThan(lastSupportedVersion) {
+		return &backend.QueryDataResponse{}, fmt.Errorf("support for elasticsearch versions after their end-of-life (currently versions < 7.10) was removed")
+	}
+
+	if len(req.Queries) == 0 {
+		return &backend.QueryDataResponse{}, fmt.Errorf("query contains no queries")
```
Comment (joey-grafana) [inline pkg/tsdb/elasticsearch/elasticsearch.go:51]: We can probably improve this message in the future! (it's been there for a while and is not very remedial)

**Review by joey-grafana — APPROVED** [review]
LGTM 👍

File: pkg/tsdb/elasticsearch/elasticsearch.go (hunk 2 above)
Comment (gabor) [reply pkg/tsdb/elasticsearch/elasticsearch.go:51]: hehe, there's a certain charm to the recursive nature of "query contains no queries" 😁

**Review by gabor — APPROVED** [review]
LGTM!

**Review by Elfo404 — APPROVED** [review]
🚀

**Review by josiahg — APPROVED** [review]
LGTM!

**Review by achatterjee-grafana — APPROVED** [review]
Approving doc changes.

**Comment by aavileli** [comment]
Elasticsearch datasource for aws opensearch 1.2 does not work any more which is based on current 7.10.2 elastic version

```bash
version	
number	"7.10.2"
build_flavor	"oss"
build_type	"tar"
build_hash	"unknown"
build_date	"2022-02-10T10:09:10.463888Z"
build_snapshot	false
lucene_version	"8.7.0"
minimum_wire_compatibility_version	"6.8.0"
minimum_index_compatibility_version	"6.0.0-beta1"
```

**Comment by ivanahuckova** [comment]
We have [opensearch data source](https://grafana.com/grafana/plugins/grafana-opensearch-datasource/) that is built for using opensearch.

**Comment by gabor** [comment]
hi @aavileli to connect to an openearch-database, we recommend using the opensearch datasource, as @ivanahuckova mentioned above.

if you can reproduce the problem with an elasticsearch 7.10.x database, please open a new issue and describe in detail what does not work, and what was the last grafana version where it worked.

**Comment by Lyky35** [comment]
whomever came up with this was stupid. Query syntax hasn't changed in any significant way to mandate removal of support from 7.X pre 7.10 versions...

**Comment by cadirol** [comment]
why remove support for deprecated Elastic Version? makes no sense at all!
Nothing changed in Elastic <7.10

**Comment by ustc093** [comment]
image

**Comment by ustc093** [comment]
OpenSearch 支持Elasticsearch 7.0+

**Comment by ustc093** [comment]
OpenSearch 支持Elasticsearch 7.0+ image

**Comment by gabor** [comment]
hi @ustc093 , i'm sorry i do not understand what you wrote. if it's related to a problem with the opensearch-datasource, please open an issue at https://github.com/grafana/opensearch-datasource

**Comment by callumj** [comment]
I am curious what brought about this change? ES hasn't changed basic query API for some time, so unless it is slowly down the development of Grafana why can't Grafana just support any version as is and if there is any problem querying it then that is on the user to address?

**Comment by voiprodrigo** [comment]
Not all organizations can keep ES installation fresh and current. That's their risk, but it happens a lot due to many reasons. So this should not have been removed from Grafana as it causes impact on observability for no really good reason. If all support for EOL software would be removed everywhere, we'd all be in a very bad situation.

**Pull request merged**